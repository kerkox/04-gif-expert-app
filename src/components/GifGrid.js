import React, { useState, useEffect } from 'react'
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

  const [images, setImages] = useState([]);

  useEffect(() => {
    getGifs();
  }, [])

  const getGifs = async () => {
    const url = `https://api.giphy.com/v1/gifs/search?q=Mr Robot&api_key=C9S94L0ayPAzoLwbX5GSZdz99ML6h5Rt&limit=10`
    // const url = `https://api.giphy.com/v1/stickers/search?q=Mr Robot&api_key=C9S94L0ayPAzoLwbX5GSZdz99ML6h5Rt&limit=10`
    const resp = await fetch(url);
    const { data } = await resp.json();

    const gifs = data.map(img => {
      return {
        id: img.id,
        title: img.title,
        url: img.images?.downsized_medium.url
      }
    })
    setImages(gifs)
  }

  return (
    <div>
      <h3>{category}</h3>
      {
        images.map(img => (
          <GifGridItem
            key={img.id} {...img} />
        ))
      }
    </div>
  )
}
