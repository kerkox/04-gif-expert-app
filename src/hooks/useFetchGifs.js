import { useEffect, useState } from "react"
import { getFetchGifs } from "../helpers/getGifs";

export const useFetchGifs = (category) => {
    const [state, setState] = useState({
        data: [],
        loading: true
    })

    useEffect(() => {
        getFetchGifs(category)
            .then(imgs => {

                setState({
                    data: imgs,
                    loading: false
                })

            })
    }, [category]);


    return state;
}