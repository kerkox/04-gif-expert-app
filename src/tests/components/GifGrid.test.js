import { shallow } from "enzyme"
import '@testing-library/jest-dom';
import { GifGrid } from "../../components/GifGrid"
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en <GifGrid/>', () => {
    const category = 'Mr Robot';

    test('Debe de hacer match con el snapshot', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })
        const wrapper = shallow(<GifGrid category={category} />)
        expect(wrapper).toMatchSnapshot();
    })

    test('debe de mostrar itesm cuando se cargan imagenes useFetchGifs', () => {
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cosa/img1.jpg',
            title: 'pruebas'
        },
        {
            id: 'ABC123',
            url: 'https://localhost/cosa/img1.jpg',
            title: 'pruebas 2'
        },
        ]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })
        const wrapper = shallow(<GifGrid category={category} />)
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    })


})
