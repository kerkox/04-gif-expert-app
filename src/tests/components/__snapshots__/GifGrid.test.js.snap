// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid/> Debe de hacer match con el snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Mr Robot
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en <GifGrid/> debe de mostrar itesm cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Mr Robot
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="pruebas"
      url="https://localhost/cosa/img1.jpg"
    />
    <GifGridItem
      id="ABC123"
      key="ABC123"
      title="pruebas 2"
      url="https://localhost/cosa/img1.jpg"
    />
  </div>
</Fragment>
`;
